piff/stats.py

```diff
@@ -0,0 +1,264 @@
+              ra_units='deg', dec_units='deg', logger=None):
+        """
+
+        :param stars:       A list of StarData instances.
+        :param logger:      A logger object for logging debug info. [default: None]
+        :param min_sep:     Minimum separation (in arcmin) for pairs
+        :param max_sep:     Maximum separation (in arcmin) for pairs
+        :param bin_size:    Logarithmic size of separation bins.
+        :param sky_coordinates: Are we in sky coordinates or otherwise? [default: False]
+        :param sep_units:   I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Sets the units of separation in min_sep, max_sep
+        :param ra_units:    I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Says what the units of the ra coordinate is.
+        :param dec_units:   I think only matters if in sky_coordinates.
+                            Can be 'rad', 'arcmin', 'deg', etc.
+                            Says what the units of the dec coordinate is.
+
+        note: assumes coord[:,0] = ra, coord[:,1] = dec or x/y
+
+        :returns: logr, rho1, rho2, rho3, rho4, rho5 correlation functions
+
+
+        Notes
+        -----
+        From Jarvis:2015 p 10, eqs 3-18 - 3-22.
+        e = e_psf ; de = e_psf - e_model
+        T is size
+
+        rho1 = < de* de >
+        rho2 = < e* de >  (in the rowe paper this is < e* de + de* e >
+        rho3 = < (e* dT / T) (e dT / T) >
+        rho4 = < de* (e dT / T) >
+        rho5 = < e* (e dT / T) >
+
+        We calculate these quantities using treecorr
+
+        also note that for gN = gNr + i gNi = gN1 + i gN2:
+        xi.xip[k] += g1rg2r + g1ig2i;       // g1 * conj(g2)
+        xi.xip_im[k] += g1ig2r - g1rg2i;
+        xi.xim[k] += g1rg2r - g1ig2i;       // g1 * g2
+        xi.xim_im[k] += g1ig2r + g1rg2i;
+
+        so since we probably really want, e.g. 0.5 < e* de + de* e >, then we
+        can just use xip.
+
+        """
+        from .gaussian_model import Gaussian
+        import treecorr
+
+        hsm = Gaussian()
+
+        # measure moments with Gaussian on image
+        if logger:
+            logger.info("Measuring Stars")
+        shapes_truth = np.array([ hsm.fitStar(star).getParameters() for star in stars ])
+        # from stars get positions
+        if logger:
+            logger.info("Getting Star Positions")
+        positions = np.array([ self.interp.getStarPosition(star) for star in stars ])
+        # generate the model stars
+        if logger:
+            logger.info("Generating Model Stars")
+        # stars_model = []
+        # for position in positions:
+        #     print(position)
+        #     stars_model.append(self.generate_star(position))
+        stars_model = np.array([ self.generate_star(position) for position in positions ])
```

Comment:
Right.  OK.  So here (assuming you agree with my proposal to sever this class from PSF), you would use `self.psf.generate_star(position)`.  

Although, I'm particularly fond of the name `generate_star` for that method.  I think the concept we want here is just drawing itself at a position.  Not generating a star, which is a little nonsensical.  This is an image of the PSF, not a star.  The fact that you wrap this in a StarModel is just an implementation detail so you could use the Gaussian class  so measure the shape.

But in general, I think the method in PSF should be:
```
psf.drawImage(position, offset)
```
where `offset` should default to (0,0), but would allow the user to offset the nominal center of the model from the center of the image if they want.

Then here, you could wrap that in a StarData object if you want, where the position to use in the constructor would be `image_pos = image.trueCenter() + offset`.